Package compatibility check target fails to parse

```
/repos/tarball-update/src/roslyn/artifacts/source-build/self/src/artifacts/obj/Microsoft.CodeAnalysis/Microsoft.CodeAnalysis.csproj.nuget.g.targets(11,5): error MSB4024: The imported project file "/repos/tarball-update/src/roslyn/artifacts/source-build/self/package-cache/system.text.encoding.codepages/7.0.0-rc.1.22403.8/buildTransitive/netcoreapp2.0/System.Text.Encoding.CodePages.targets" could not be loaded. An error occurred while parsing EntityName. Line 4, position 216. [/repos/tarball-update/src/roslyn/artifacts/source-build/self/src/src/Compilers/Core/Portable/Microsoft.CodeAnalysis.csproj]
```

Looking around, I think this may be related to https://github.com/dotnet/runtime/pull/72518.  It appears the `;` in the warning message is getting interpreted as a line break therefore making the `&lt` invalid.

The NuGet package in my build shows the following:
```
<Project InitialTargets="NETStandardCompatError_System_Text_Encoding_CodePages_net6_0">
  <Target Name="NETStandardCompatError_System_Text_Encoding_CodePages_net6_0"
          Condition="'$(SuppressTfmSupportBuildWarnings)' == ''">
    <Warning Text="System.Text.Encoding.CodePages 7.0.0-rc.1.22403.8 doesn't support $(TargetFramework) and has not been tested with it. Consider upgrading your TargetFramework to net6.0 or later. You may also set &lt
SuppressTfmSupportBuildWarnings&gt
true&lt
/SuppressTfmSupportBuildWarnings&gt
in the project file to ignore this warning and attempt to run in this unsupported configuration at your own risk." />
  </Target>
</Project>
```

cc @ViktorHofer

--- a/eng/packaging.targets
+++ b/eng/packaging.targets
@@ -200,7 +200,7 @@
 <![CDATA[<Project InitialTargets="$(_NETStandardCompatErrorFileTarget)">
   <Target Name="$(_NETStandardCompatErrorFileTarget)"
           Condition="'%24(SuppressTfmSupportBuildWarnings)' == ''">
-    <Warning Text="$(PackageId) $(PackageVersion) doesn't support %24(TargetFramework) and has not been tested with it. Consider upgrading your TargetFramework to %(NETStandardCompatError.Supported) or later. You may also set &lt;SuppressTfmSupportBuildWarnings&gt;true&lt;/SuppressTfmSupportBuildWarnings&gt; in the project file to ignore this warning and attempt to run in this unsupported configuration at your own risk." />
+    <Warning Text="$(PackageId) $(PackageVersion) doesn't support %24(TargetFramework) and has not been tested with it. Consider upgrading your TargetFramework to %(NETStandardCompatError.Supported) or later. You may also set &lt%3BSuppressTfmSupportBuildWarnings&gt%3Btrue&lt%3B/SuppressTfmSupportBuildWarnings&gt%3B in the project file to ignore this warning and attempt to run in this unsupported configuration at your own risk." />
   </Target>
 </Project>]]>
       </_NETStandardCompatErrorFileContent>

